### PR TITLE
feat: 404 page

### LINKED ISSUE #910: Feature: Insights need custom 404 Not-found page
### Type of feature

🍕 Feature

### Current behavior

Our current authenticated routes are yet to redirect unauthenticated users to a 404 not found page, and a part of the reason is we don't have a custom 404-page design...  

### Suggested solution

@getaheaddev can sketch something out that can be reused across the application to handle authenticated routes

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by foxyblocks**
## Description

Adds a new custom 404 page that shows dev cards from various users. 

### TODO

- [ ] Needs the `users/top` api endpoint to support more than 10 users, cc @OgDev-01  @brandonroberts is there a ticket for this?

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Closes #910 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/insights/assets/555044/61a5cbb4-d3e6-4394-bedb-2e203af2891d

CleanShot 2023-07-19 at 15 51 11@2x

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

confused-superbowl-gif-by-republic-records

### CODE REVIEW

File: pages/404.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        }}
+      >
+        <div className="grid justify-center place-content-start py-7 px-3 md:justify-start">
+          <HeaderLogo withBg={false} />
+        </div>
+        <main className="grid md:grid-cols-2 place-content-center py-6">
+          <div className="text-center px-6 relative z-10">
+            <h1 className="text-7xl font-bold mb-2">404</h1>
+            <div className="text-2xl mb-2">uh oh page not found</div>
+            <div className="mb-2">while you&apos;re here, you can check out some of our amazing users!</div>
```
Comment (bdougie) [inline pages/404.tsx:62]:
```suggestion
            <div className="mb-2">while you&apos;re here, you can check out some of our amazing contributors!</div>
```

small change to "contributors" per this discussion https://github.com/open-sauced/insights/pull/1347#issuecomment-1637247449

**Comment by bdougie** [comment]
@foxyblocks This looks great. One ask is the card page allows key navigation and enhances the accessibility. Are we able to key bindings for navigating different cards to this PR.

**Comment by foxyblocks** [comment]
> @foxyblocks This looks great. One ask is the card page allows key navigation and enhances the accessibility. Are we able to key bindings for navigating different cards to this PR. 

It currently supports the "left" arrow key for navigation to coincide with the arrow button. Did you want another key??

**Comment by bdougie** [comment]
> It currently supports the "left" arrow key for navigation to coincide with the arrow button. Did you want another key??

Oh. I was going up and down. Lol

Perhaps we can include "down" and have it do the same as "left." Not sure if that is help for accessibility but would be for me.

**Comment by brandonroberts** [comment]
@foxyblocks the last build failed with some prettier changes, will you take a look?

**Review by brandonroberts — APPROVED** [review]
Changes are good. Just needs to build/lint fix.

**Review by OgDev-01 — APPROVED** [review]

**Pull request merged**